apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/RdaServerJobIT.java

```diff
@@ -58,8 +63,20 @@ public class RdaServerJobIT extends MinioTestContainer {
   private final McsClaimTransformer mcsTransformer =
       new McsClaimTransformer(clock, MbiCache.computedCache(hasherConfig));
 
-  /** only need a single instance of the S3 client. */
-  private static final S3Client s3Client = createS3MinioClient();
+  /** Configuration settings to connect to localstack container. */
+  private S3ClientConfig s3ClientConfig;
+  /** Factory to create clients connected to localstack container. */
+  private S3ClientFactory s3ClientFactory;
+  /** A client connected to the localstack container for use in test methods. */
+  private S3Client s3Client;
+
+  /** Populates S3 related fields based on localstack container. */
+  @BeforeEach
+  void initializeS3RelatedFields() {
+    s3ClientConfig = LocalStackS3ClientFactory.createS3ClientConfig(localstack);
+    s3ClientFactory = new AwsS3ClientFactory(s3ClientConfig);
+    s3Client = s3ClientFactory.createS3Client();
+  }
```

Comment:
I think we could move these test variables and setup into the abstract class if all the localStack tests will use the same setup; I did this in the ServerRequiredTest base class for instance.

Reply:
Not unless we make that class S3 specific.

Reply:
I'll take a look tomorrow.  Maybe I'll introduce an S3 specific subclass.  I'm not a fan of this base class pattern anyway and might revisit it in BFD-2181.

Reply:
I have Added AbstractLocalStackS3Test and removes LocalStackS3ClientFactory.